flink-runtime/src/main/java/org/apache/flink/runtime/rpc/jobmaster/JobMaster.java

```diff
@@ -59,62 +42,52 @@
  * It offers the following methods as part of its rpc interface to interact with the JobMaster
  * remotely:
  * <ul>
- *     <li>{@link #registerAtResourceManager(String)} triggers the registration at the resource manager</li>
  *     <li>{@link #updateTaskExecutionState(TaskExecutionState)} updates the task execution state for
  * given task</li>
  * </ul>
  */
 public class JobMaster extends RpcEndpoint<JobMasterGateway> {
-	/** Execution context for future callbacks */
-	private final ExecutionContext executionContext;
-
-	/** Execution context for scheduled runnables */
-	private final ScheduledExecutorService scheduledExecutorService;
-
-	private final FiniteDuration initialRegistrationTimeout = new FiniteDuration(500, TimeUnit.MILLISECONDS);
-	private final FiniteDuration maxRegistrationTimeout = new FiniteDuration(30, TimeUnit.SECONDS);
-	private final FiniteDuration registrationDuration = new FiniteDuration(365, TimeUnit.DAYS);
-	private final long failedRegistrationDelay = 10000;
 
 	/** Gateway to connected resource manager, null iff not connected */
 	private ResourceManagerGateway resourceManager = null;
 
-	/** UUID to filter out old registration runs */
-	private UUID currentRegistrationRun;
-
 	/** Logical representation of the job */
-	private JobGraph jobGraph;
-	private JobID jobID;
+	private final JobGraph jobGraph;
+	private final JobID jobID;
 
 	/** Configuration of the job */
-	private Configuration configuration;
-	private RecoveryMode recoveryMode;
+	private final Configuration configuration;
+	private final RecoveryMode recoveryMode;
 
 	/** Service to contend for and retrieve the leadership of JM and RM */
-	private HighAvailabilityServices highAvailabilityServices;
+	private final HighAvailabilityServices highAvailabilityServices;
 
 	/** Leader Management */
 	private LeaderElectionService leaderElectionService = null;
 	private UUID leaderSessionID;
 
+	/**
+	 * The JM's Constructor
```

Comment:
Javadocs usually say what the method/constructor does and not what it is. At the moment, it just states the obvious.